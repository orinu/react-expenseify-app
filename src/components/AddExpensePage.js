import React from 'react';
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {addExpense} from '../action/expenses';


const HomePage = (props) => (
    <div>
      <h1>Add</h1>
      <ExpenseForm 
        onSubmit={(expense) => {
          props.dispatch(addExpense(expense));
          props.history.push('/')
        }}
      />
    </div>
)

export default connect()(HomePage);