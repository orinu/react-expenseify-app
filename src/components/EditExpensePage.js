import React from 'react';
import {editExpense, removeExpense} from '../action/expenses';
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';



const EditExpensePage = (props) => {
  return (
    <div>
      <ExpenseForm 
        expense={props.expense}
        onSubmit={(expense) => {
          props.dispatch(editExpense(props.expense.id, expense));
          props.history.push('/');
        }} 
      />
      <button onClick={() => {
        props.dispatch(removeExpense({ id: props.expense.id}))
        props.history.push('/');
    }}>Remove</button>
    </div>
  );
};

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
 };
};

export default connect(mapStateToProps)(EditExpensePage);
