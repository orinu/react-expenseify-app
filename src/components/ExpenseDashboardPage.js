import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFIlters'
import ExpensesSummery from './ExpensesSummary';


const ExpenseDashbordPage = () => (
    <div>
      <ExpensesSummery />
      <ExpenseListFilters />
      <ExpenseList />
     
    </div>
)

export default ExpenseDashbordPage;