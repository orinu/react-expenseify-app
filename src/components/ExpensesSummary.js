import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpense from '../selectors/expenses';
import selectExpensesTotal from '../selectors/Expenses-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal / 100 ).format('$0,0.00')
    return (
        <div>
            <h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</h1>
        </div>
    )
};

const mapStateToProps = (state) => {
    const visibleEzpenses = selectExpense(state.expenses, state.filters);
    return {
        expenseCount: visibleEzpenses.length,
        expensesTotal: selectExpensesTotal(visibleEzpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary)