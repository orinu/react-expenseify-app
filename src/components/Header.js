import React from 'react';
import { NavLink} from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../action/auth';

export const Header = ({ startLogout }) => (
    <header>
     <h1>Expend</h1>
     DashBoard - <NavLink to="dashboard" activeClassName="is-active" exact={true}>DashBoard</NavLink>
     Create - <NavLink to="/create" activeClassName="is-active" exact={true}>Create</NavLink>
     help - <NavLink to="/help" activeClassName="is-active">help</NavLink>
     <button onClick={startLogout}>Logout</button>
    </header>
)

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header);