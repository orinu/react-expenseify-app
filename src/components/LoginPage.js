import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../action/auth';

export const LoginPage = ({startLogin}) =>  (
        <div>
            <h1>Log In</h1>
            <button onClick={startLogin}>LogIn</button>
        </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);

