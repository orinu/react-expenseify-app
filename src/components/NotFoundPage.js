import React from 'react';
import {Link} from 'react-router-dom';

const NotFoundPage = () => (
    <div>
      <Link to="/"> this is from 404</Link>
    </div>
)

export default NotFoundPage;