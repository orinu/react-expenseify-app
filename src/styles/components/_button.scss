.button {
    background-color: $blue;
    color: white;
    border: none;
    padding: $s-size;
    font-size: $font-size-large;
    font-weight: 300;
    text-decoration: none;
    line-height: 1;
}

.button--link {
    background: none;
}

.button--secondary {
    background: #888;
}