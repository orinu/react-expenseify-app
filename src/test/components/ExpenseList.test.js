import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from '../../components/ExpenseList';
import expenses from '../fixture/expenses';

test('Shold render EpenseList wtih expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses}/>)
    expect(wrapper).toMatchSnapshot();
});

test('Shold render EpenseList with empty message', () => {
    const wrapper = shallow(<ExpenseList expenses={[]}/>)
    expect(wrapper).toMatchSnapshot();
});