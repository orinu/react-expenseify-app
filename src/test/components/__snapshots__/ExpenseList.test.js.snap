// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shold render EpenseList with empty message 1`] = `
<div>
  <p>
    No expenses
  </p>
</div>
`;

exports[`Shold render EpenseList wtih expenses 1`] = `
<div>
  <ExspenseListItem
    amount={196}
    createdAt={0}
    description="Gum"
    id="1"
    key="1"
    note=""
  />
  <ExspenseListItem
    amount={1000}
    createdAt={-345600000}
    description="Rent"
    id="2"
    key="2"
    note=""
  />
  <ExspenseListItem
    amount={2000}
    createdAt={345600000}
    description="CreditCard"
    id="3"
    key="3"
    note=""
  />
</div>
`;
